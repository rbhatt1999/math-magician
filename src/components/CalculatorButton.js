import React from 'react';
import PropTypes from 'prop-types';

export default class CalculatorButton extends React.PureComponent {
  render() {
    const { text, className, handleClick } = this.props;
    return (
      <button className={`btn ${className}`} type="button" onClick={handleClick}>
        {text}
      </button>
    );
  }
}

CalculatorButton.defaultProps = {
  className: '',
};

CalculatorButton.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};
