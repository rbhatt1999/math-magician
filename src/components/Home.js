const Home = () => (
  <div className="py-2">
    <h1 className="text-xl">Welcome to Math Magician</h1>
    <p className="mt-1">We help you to solve your complex math problems using our magical caculator.</p>
    <div className="py-2">
      <img
        src="https://www.wellesley.edu/sites/default/files/styles/news_refresh_hero/public/assets/dailyshot/ds_461390782.jpg?itok=jr0Buv1t"
        className="w-full"
        alt="math pic"
      />
    </div>
  </div>
);

export default Home;