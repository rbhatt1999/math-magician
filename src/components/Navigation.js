import { NavLink } from 'react-router-dom';
import '../css/styles.css';

const Navigation = () => (
  <nav className="flex justify-between container py-2 bg-white shadow">
    <h2 className="text-lg">Math Magician</h2>
    <ul className="flex gap">
      <li>
        <NavLink style={({ isActive }) => ({ color: isActive ? 'red' : '' })} end to="/math-magician">Home</NavLink>
      </li>
      <li>
        <NavLink style={({ isActive }) => ({ color: isActive ? 'red' : '' })} end to="/math-magician/calculator">Calculator</NavLink>
      </li>
      <li>
        <NavLink style={({ isActive }) => ({ color: isActive ? 'red' : '' })} end to="/math-magician/quotes">Quotes</NavLink>
      </li>
    </ul>
  </nav>
);

export default Navigation;
