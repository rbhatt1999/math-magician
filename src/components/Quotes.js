const Quotes = () => {
  const quote = 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston';
  return (
    <div className="py-2 w-full h-full flex justify-between items-center">
      <p className="text-2xl text-center">{quote}</p>
    </div>
  );
};

export default Quotes;
