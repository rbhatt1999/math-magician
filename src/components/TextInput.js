import PropTypes from 'prop-types';

const TextInput = (props) => {
  const changeHandler = () => {
    const { value } = props;
    return value;
  };

  const { value } = props;
  return <input type="text" value={value} onChange={changeHandler} />;
};

TextInput.propTypes = {
  value: PropTypes.number.isRequired,
};

export default TextInput;
