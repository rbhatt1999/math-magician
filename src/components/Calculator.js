import React from 'react';
import CalculatorButton from './CalculatorButton';
import TextInput from './TextInput';

class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="calculator">
        <div className="grid cols-4">
          <div className="span-4">
            <TextInput value={0} />
          </div>
          <CalculatorButton text="AC" />
          <CalculatorButton text="+/-" />
          <CalculatorButton text="%" />
          <CalculatorButton className="bg-orange" text="รท" />
          <CalculatorButton text="7" />
          <CalculatorButton text="8" />
          <CalculatorButton text="9" />
          <CalculatorButton className="bg-orange" text="ร" />
          <CalculatorButton text="4" />
          <CalculatorButton text="5" />
          <CalculatorButton text="6" />
          <CalculatorButton className="bg-orange" text="-" />
          <CalculatorButton text="1" />
          <CalculatorButton text="2" />
          <CalculatorButton text="3" />
          <CalculatorButton className="bg-orange" text="+" />
          <CalculatorButton className="span-2" text="0" />
          <CalculatorButton text="." />
          <CalculatorButton className="bg-orange" text="=" />
        </div>
      </div>
    );
  }
}

export default Calculator;
