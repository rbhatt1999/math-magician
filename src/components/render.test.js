import renderer from 'react-test-renderer';
import Home from './Home';
import Calculator from './Calculator';
import CalculatorButton from './CalculatorButton';
import TextInput from './TextInput';
import Quotes from './Quotes';
import Navigation from './Navigation';
import { MemoryRouter } from 'react-router-dom';

it('Home renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Calculator renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CalculatorButton renders correctly', () => {
  const handleClick = () => true;
  const tree = renderer.create(<CalculatorButton text="Submit" handleClick={handleClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('text unput renders correctly', () => {
  const tree = renderer.create(<TextInput value={43} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Quotes renders correctly', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Home renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});