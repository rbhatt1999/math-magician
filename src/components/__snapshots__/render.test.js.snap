// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  className="py-2 flex justify-between flex-wrap"
>
  <div
    className="sm:w-half"
  >
    <h1
      className="text-xl"
    >
      Let's do some math
    </h1>
    <img
      alt=""
      className="w-full mt-1"
      src="https://www.callcentrehelper.com/images/stories/2016/04/calulator-hand-760.jpg"
    />
  </div>
  <div
    className="calculator mt-1"
  >
    <div
      className="grid cols-4"
    >
      <div
        className="span-4"
      >
        <input
          onChange={[Function]}
          type="text"
          value={0}
        />
      </div>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn span-2"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`CalculatorButton renders correctly 1`] = `
<button
  className="btn "
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;

exports[`Home renders correctly 1`] = `
<div
  className="py-2"
>
  <h1
    className="text-xl"
  >
    Welcome to Math Magician
  </h1>
  <p
    className="mt-1"
  >
    We help you to solve your complex math problems using our magical caculator.
  </p>
  <div
    className="py-2"
  >
    <img
      alt="math pic"
      className="w-full"
      src="https://www.wellesley.edu/sites/default/files/styles/news_refresh_hero/public/assets/dailyshot/ds_461390782.jpg?itok=jr0Buv1t"
    />
  </div>
</div>
`;

exports[`Home renders correctly 2`] = `
<nav
  className="flex justify-between container py-2 bg-white shadow"
>
  <h2
    className="text-lg"
  >
    Math Magician
  </h2>
  <ul
    className="flex gap"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Quotes renders correctly 1`] = `
<div
  className="py-2 w-full h-full flex justify-between items-center"
>
  <p
    className="text-2xl text-center"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
  </p>
</div>
`;

exports[`text unput renders correctly 1`] = `
<input
  onChange={[Function]}
  type="text"
  value={43}
/>
`;
